### PR TITLE
Log check/eval panics for telemetry

### PR DESCRIPTION
**Pull request by justicz**
We will want to know if/when these panics happen. Test coverage in `eval_test.go`'s `TestPanic`.

### CODE REVIEW

**Comment by winder** [comment]
How often do these happen? We don't want to flood telemetry with too many events. Also, does it need to be an error instead of a warning? I think we attach a log history with errors, and maybe that isn't necessary.

**Comment by justicz** [comment]
The goal is that these recovered panics never happen -- they're what the [TEAL fuzzer](https://github.com/algorand/tealfuzz) is trying to find. In terms of severity we should definitely consider it more of an "error" than a "warning", but we probably don't need to include logs (also, the error message itself includes a stack trace)

Should I make it a warning?

**Comment by algobolson** [comment]
@winder it's a "this should never happen" level of error that means we had a programming bug. pretty sure it's exactly the thing we should be reporting to ourselves.

**Comment by winder** [comment]
Ok, error sounds right

File: data/transactions/logic/eval.go [hunk 1]
```diff
@@ -180,6 +181,7 @@ func Eval(program []byte, params EvalParams) (pass bool, err error) {
 					errstr += sb.String()
 				}
 			}
+			logging.Base().Errorf("recovered panic in Eval: %s", errstr)
```
Comment (algobolson) [inline data/transactions/logic/eval.go]: you should include the panic value x in these. see `func (pe PanicError) Error() string`

Comment (justicz) [reply data/transactions/logic/eval.go]: thanks, I'm just `Errorf`ing the `PanicError` now

**Comment by justicz** [comment]
An error here indicates a bug in _our_ code, not just in the TEAL script's code. A recovered panic indicates a programming mistake in our interpreter/checker.

**Pull request merged**